toolchain/driver/compile_subcommand.cpp

```diff
@@ -694,9 +686,12 @@ auto CompilationUnit::RunCodeGenHelper() -> bool {
     llvm::raw_fd_ostream output_file(output_filename, ec,
                                      llvm::sys::fs::OF_None);
     if (ec) {
-      *driver_env_->error_stream << "error: could not open output file '"
-                                 << output_filename << "': " << ec.message()
-                                 << "\n";
+      CARBON_DIAGNOSTIC(CompileOutputFileOpenError, Error,
+                        "could not open output file `{0}`: {1}", std::string,
+                        std::string);
+      driver_env_->filename_emitter.Emit("", CompileOutputFileOpenError,
+                                         output_filename.str().str(),
```

Comment:
FWIW, this is somewhat my motivation for not using the filename emitter for command errors -- there *is* a filename that is relevant here and even passed in, but it isn't passed as the location. That makes sense in the context, but I think is surprising because this is using the filename location with a special value.

Reply:
I'm maybe on the other side here... Added a TODO to rewrite the diagnostic, I'd been trying to keep the change minimal. :)

Reply:
(As discussed in person, yeah, this diagnostic might end up rewritten. Nothing to do here.)